presto-main/src/test/java/com/facebook/presto/type/TestJsonOperators.java

```diff
@@ -251,8 +286,57 @@ public void testCastFromTimestamp()
         assertFunction("CAST(from_unixtime(1) AS JSON)", JSON, "\"" + sqlTimestamp(1000).toString() + "\"");
     }
 
+    @Test
+    public void testCastWithJsonParse()
```

Comment:
These test cases looks familiar, are they collected from `TestArrayOperators` and `TestMapOperators`?

These legacy test cases are not complete and a few will be deprecated in https://github.com/prestodb/presto/pull/8217. However, I think here the purpose for test is not to have a comprehensive testing, but more like a smoke test to make sure everything is working reasonable .

I feel it might help to just use a subset of new tests in https://github.com/prestodb/presto/pull/8217  ? Or we can unify them in a separate commit later (after both PRs are merged).  Maybe the latter is better since https://github.com/prestodb/presto/pull/8217 also requires some changes